numpy/core/src/umath/ufunc_object.c

```diff
@@ -516,6 +517,55 @@ _apply_array_wrap(
 }
 
 
+/*
+ * TODO: Refactor this to be reused by ResultType, which includes the same.
```

Comment:
Can you include a filename / commit / linenumber of the bit of ResultType that uses this?